plugins/woocommerce/client/blocks/assets/js/atomic/blocks/product-elements/image/block.json

```
@@ -48,7 +50,7 @@
 		"woocommerce/product-template",
 		"core/post-template"
 	],
-	"usesContext": [ "query", "queryId", "postId" ],
+	"usesContext": [ "imageId", "postId", "query", "queryId" ],
```

In testing steps you mention:

> Add a new context provider to the providesContext object:
> `"woocommerce/product-image/imageId": "imageId"`

so either testing steps should be updated or update `usesContext`:

```suggestion
	"usesContext": [ "woocommerce/product-image/imageId", "postId", "query", "queryId" ],
```

---

What do you think? The [docs](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-context/#providing-block-context) recommend to include a namespace, which I think makes sense. However, none of our contexts include a namespace, so that's kinda inconsistent…

---

Most of them don't but some of them do:
- [Add to Cart](https://github.com/woocommerce/woocommerce/blob/91fbd4f72587ac8ff28cf61038a3ec4b07bc6e22/plugins/woocommerce/client/blocks/assets/js/blocks/add-to-cart-with-options/block.json#L14)
- [Product Reviews](https://github.com/woocommerce/woocommerce/blob/2b22912d1ff83bc3ec94ebf457907909cba5a6ee/plugins/woocommerce/client/blocks/assets/js/blocks/product-reviews/inner-blocks/reviews-pagination/block.json#L27)

TBH most context we provide is pretty unique and most likely we will avoid name clashes (e.g. `fullScreenOnClick` in Product Gallery) while some are less specific. `imageId` seems more common BUT... I **wouldn't** namespace it. I think it has potential to become more universal context similar to `query` or `queryId` so basically Product Image could be compatible with other blocks displaying images. In other words, I image Product Image / Image / Cover could be at some point used inside Product Gallery (or elsewhere) in order to display specific content provided via `imageId` context.